### ISSUE TITLE
Project Review

### ISSUE THREAD

**Issue by jordanwillis** [tepope/pwp-capstones#1]
### Criteria 1: Valid Python Code
* _Score Level:_ 4/4
* _Comment(s):_ Great job crafting not only valid, but nice looking python!  I really liked your overall style and ability to take advantage of built in functions and syntax for nice compact code.
 
### Criteria 2: Implementation of Project Requirements
* _Score Level:_ 4/4
* _Comment(s):_ All of the necessary requirements were implemented perfect.  Below is a quick note that I jotted down during the review.

I would actually recommend not returning an integer from `get_average_sentence_length` function because you won't get as accurate of results.  In general with data analysis you will want as much precision as possible, so only deal with integers if its a specific requirement.  Here is what is causing the result to always be an integer.

``` python
average_sentence_length = (len(text) // len(sentences))
```
 
### Criteria 3: Software Architecture
* _Score Level:_ 4/4
* _Comment(s):_ Your implementation was broken up into appropriate classes and functions, nice job!
 
### Criteria 4: Uses Python Language Features
* _Score Level:_ 4/4
* _Comment(s):_ As I mentioned earlier, your use of built-in language features and syntax was excellent!  For example, leveraging `join`, a nested loop, and other functions in your `prepare_text` to made for a very effective and easy to understand 1-line implementation (whereas many other students would use 5 - 10 lines).  This was just one specific example, but all of your code was great!

### Criteria 5: Produces Accurate Output
* _Score Level:_ 4/4
* _Comment(s):_ Congratulations,, you deduced the correct murderer (and the rest of your output was correct as well)!
 
### Overall Score: 20/20
Overall you did an absolutely fantastic job on this project which made it a pleasure to review.

Keep up the great work!

**Comment by tepope** [comment]
Thank you for the feedback! Can you comment on the integer suggestion a
little more? What would you recommend returning for the average? a float?

**Comment by tepope** [comment]
Also, can you suggest a way to make the frequency_comparison a little more readable? It feels very verbose to me.

**Comment by jordanwillis** [comment]
Yup, that is correct!  I would return a float just so you don't lose out of that extra bits of numerical information.  Typically with any sort of data analysis you will want more precision (e.g. decimal places) than what an integer can provide.  So just using the single slash was what I had in mind.

``` python
average_sentence_length = len(text) / len(sentences)
```

Actually, your implementation was pretty concise.  It's just a tough problem to solve (e.g. frequency_comparison) and requires all the looping, etc.  Here's mine if you are interested.

``` python
def frequency_comparison(table1, table2):
    appearances = 0
    mutual_appearances = 0
    for key in table1:
        if key in table2:
            value1 = table1.get(key)
            value2 = table2.get(key)
            if value2:
                if value1 > value2:
                    mutual_appearances += value2
                    appearances += value1
                else:
                    mutual_appearances += value1
                    appearances += value2
            else:
                appearances += value1
        else:
            appearances = table1.get(key)

    keys_not_in_table1 = [key for key in table2 if key not in table1]
    for key in keys_not_in_table1:
        appearances += table2.get(key)

    return mutual_appearances / appearances
```

**Comment by tepope** [comment]
Thank you for the feedback. Seeing your `frequency_comparison` helped a lot!
